pytorch_lightning/loggers/neptune.py

```diff
@@ -318,7 +318,7 @@ def log_text(self, log_name: str, text: str, step: Optional[int] = None) -> None
             text: The value of the log (data-point).
             step: Step number at which the metrics should be recorded, must be strictly increasing
         """
-        self.experiment.log_text(log_name, text, step=step)
+        self.experiment.log_text(log_name, text)
```

Comment:
should it be like:
```python
if step is None:
    self.experiment.log_text(log_name, text)
else:
    self.experiment.log_text(log_name, x=step, y=text)
```
??
ref: https://neptune-client.readthedocs.io/en/docs-notebooks-api/technical_reference/experiment.html#neptune.experiments.Experiment.log_text

Reply:
The neptune experiment requires that the `x` argument must be strictly increasing at each invocation (when `y` is set), and we don't prevent/check that, so would we really expose that parameter? In `NeptuneLogger.log_metrics()` we already ignore the `step` argument for the same reason and, if the user want to specify `y` he can still directly call `NeptuneLogger.experiment.log_text()`

Reply:
well, the above condition satisfies both the conditions though, also matches with other methods like `log_image`, `log_metric`. This way user can just use the available API `self.logger.log_text()` from lightning instead of `self.logger.experiment.log_text()` and going over Neptune docs.

Plus I think we should have a simple `kwargs` argument in each of these methods which will directly be passed to the corresponding native Neptune method API, allowing all the available functionality without any restrictions. For eg `log_image` in Neptune as `description`, and `timestamp`, we can make these available to the user out of the box, also updating the docs of each method with some relevant links.

Reply:
I failed to see how that if can satisfy the condition on a strictly increasing `x` (and so `step`). In lines 362-363 of `pytorch_lightning/loggers/neptune.py` is written that:
```
        for key, val in metrics.items():
            # `step` is ignored because Neptune expects strictly increasing step values which
            # Lighting does not always guarantee.
            self.log_metric(key, val)
```

About the `kwargs` argument: it sounds good, but maybe is out of scope of this pr?

Reply:
`log_metrics` is used internally by lightning and since `global_step` is used in place of the `step` which is either increases or stays the same, that's why it's disabled there. But since `log_text` is user specific API and it not used internally, one can simply create a counter which is strictly increasing at every step and use that instead. By default, the step will be None, but what I am suggesting here is that we should not disable such a use-case altogether.

> About the kwargs argument: it sounds good, but maybe is out of scope of this or?

yes ofcourse not in the one, a separate one would be better.

Reply:
Done, changed as you suggested